README.md

```diff
@@ -126,17 +150,34 @@ The team worked with the GitHub task management system for managing the workload
 ![](https://webteic.info/images/PMScore-7.jpg)
 > Screenshot: Team development workload.
 
-Details here of how the system works in the final version.
+The tasks were agreed with each person on the team & setup of the kanban board proceeded with column limits to ensure a smooth process.
 
-- Key implementation decision 1
-- Key implementation decision 2
-- Key implementation decision 3
-- Key implementation decision 4
+During the development we would do unit tests at the same time of developing each task, a continuous cycle to reduce the number of potential bugs that could appear later on.
+
+Branches would be used with GitHub for each persons changes before being merged later into the master.
+
+The team encountered a major bottleneck later in the project lifecycle and the original scope needed to be reshaped. During a few team meetings, we decided to reduce the number of tasks and combine certain processes. This would enhance the project while also making more attainable to achieve the goals set out. 
+
+UML Diagrams were updated to account for this with the User -> TaskManager, ProjectManager & UserManager -> with the improved StateManager. This streamlined the structure of the project and was one of the most vital implementation decisions made.
+
+The bottlenecks were also removed with the team effort on additional reviewing & fixing, as well removing unnecessary tasks no longer needed with the improved structure. 
+
+![](https://webteic.info/images/PMScore-7-1.jpg)
+> Screenshot: Team development workload improvements.
+
+During team meetings, everyone had their voice to address any issues and the team began to gel as one unit to reach the goals set out. Each person showed their own leadership abilities, support & working as one to deliver the project.
 
 ###  Testing
 
-Details here
+Rigorous testing (unit tests) was done from the outset with each task & classes. While this did slow down the overall development, it was better this way to allow a cleaner codebase that was not riddled with bugs.
 
-## Implementation Decisions
```

Comment:
In # Implementation Decisions it would be good to add a sentence or so about the 4 core pillars of object-oriented programming: inheritance, abstraction, encapsulation, and polymorphism.

Additionally, briefly mention how JSON handles data, also how our work was tracked on the GitHub Kanban board and maybe a challenge and solution: e.g. a bottleneck in this and that class was handled and the UML diagram updated based on that.